lib/protoflow-blocks/src/blocks/sys/write_file.rs

```diff
@@ -60,10 +60,25 @@ impl WriteFile {
 }
 
 impl Block for WriteFile {
-    fn execute(&mut self, _runtime: &dyn BlockRuntime) -> BlockResult {
-        while let Some(_message) = self.input.recv()? {
-            unimplemented!() // TODO
+    fn execute(&mut self, runtime: &dyn BlockRuntime) -> BlockResult {
+        use std::io::prelude::Write;
+
+        runtime.wait_for(&self.path)?;
+
+        let Some(path) = self.path.recv()? else {
+            return Ok(());
+        };
+        let mut file = std::fs::OpenOptions::new()
+            .create(true)
+            .append(true)
+            .open(path)?;
+
+        while let Some(message) = self.input.recv()? {
+            file.write_all(&message)?;
         }
+
+        drop(file);
```

Comment:
Is there a reason to call it here explicitly? I remember this was explicitly stated in some blocks only for messages on the input ports.

Reply:
No reason or requirement really, simply feels appropriate to close the file handle as soon as you get the chance.

Worst case, the `self.input.close()` below blocks for a while, either with the current `mpsc` transport which acquires a mutex, or some hypothetical transport which does disk/network io to signal closing the input.